### PR TITLE
update MD files for software dev course

### PR DESCRIPTION
**Pull request by SissiFeng**

### CODE REVIEW

File: docs/courses/software-dev/4.1-vscode-setup.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+   ```
+3. Copy the public key to your remote server:
+   ```bash
+   ssh-copy-id username@hostname
+   ```
+4. Use VS Code's Remote Explorer to connect to the remote server using SSH.
+
+**Video Tutorial**: [VS Code Remote Development with SSH](https://www.youtube.com/watch?v=miyD4c1dnTU&pp=ygUjVlMgQ29kZSBSZW1vdGUgRGV2ZWxvcG1lbnQgd2l0aCBTU0g%3D)
+
+### Configuring Black Formatter
```
Comment (sgbaird) [inline docs/courses/software-dev/4.1-vscode-setup.md:84]: I still need to finish my review, but I'll come back to this

**Review by sgbaird — CHANGES_REQUESTED** [review]

File: docs/courses/software-dev/4.1-vscode-setup.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+### Setting Up VS Code
+
+First, you will download and install Visual Studio Code (VS Code), a lightweight and powerful editor. Then, you'll learn how to configure it for efficient Python development.
+
+#### Steps:
+
+1. Download and install VS Code from the official site: [VS Code Download](https://code.visualstudio.com/).
+2. Open VS Code and get familiar with the interface, including the command palette, sidebar, and settings menu.
+
+**Video Tutorial**: [Getting Started with VS Code](https://www.youtube.com/watch?v=VqCgcpAypFQ)
```
[the remark this answers is not in the document]
Comment (sgbaird) [reply docs/courses/software-dev/4.1-vscode-setup.md:28]: Feedback: original one is more beginner friendly

File: docs/courses/software-dev/4.1-vscode-setup.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+**Video Tutorial**: [Best VS Code Extensions for Python](https://www.youtube.com/watch?v=fj2tuTIcUys&pp=ygUiQmVzdCBWUyBDb2RlIEV4dGVuc2lvbnMgZm9yIFB5dGhvbg%3D%3D)
+
+### Using SSH for Remote Development
+
+VS Code supports remote development, allowing you to write and run code on a remote machine as if it were local.
+
+#### Steps:
+
+1. Install the **Remote - SSH** extension from the marketplace.
+2. Set up SSH on your local machine by generating SSH keys:
```
[the remark this answers is not in the document]
Comment (sgbaird) [reply docs/courses/software-dev/4.1-vscode-setup.md:72]: Usecase: Raspberry Pi ssh with VS Code extension. I'll get you an RPi 5 and a screen. Maybe mention HPC as a common example in physical sciences

**Comment by SissiFeng** [comment]
> I forget if it was this one or the other PR that I still couldn't compile the docs with. I was able to for some of the other PRs though

I will integrate your changes from this PR into the original document and then open a new PR.

**Review by sgbaird — APPROVED** [review]

**Pull request merged**